### DISCUSSION TITLE
Running a script as a first step?

### DISCUSSION THREAD

**Discussion by papanito** [twpayne/chezmoi#2262] [Q&A] [answered]
As far as I have seen, when  you do a `chezmoi apply` it first places the dotfiles and then runs the scripts. Is there a way to run a script before everything?

I want to use keyring to populate some tokens, so as a first I would like to add missing stuff to keyring by having `run_once_populate.keyring.sh.tmpl`

```
{{ if eq .chezmoi.os "linux" -}}
#!/bin/sh
for service in chezmoi github gitlab b2_account_id b2_account_secret b2_account_key b2_repository b2_account_secret; do
    chezmoi secret keyring get --service $service --user=papanito >/dev/null || echo "Value for '$service' in keyring missing"; chezmoi secret keyring set --service $service --user=papanito
done
{{ end -}}
```

Once this script is run I can use keyring, for example in `dot_gitconfig.tmpl`

```
[github]
    user = {{ .githubuser | quote }}
    token = {{ keyring "github" .githubuser | quote }}
```

**Comment by twpayne** [comment, marked as answer]
[`run_before_` scripts](https://www.chezmoi.io/reference/application-order/) do this.

**Reply by papanito** [reply]
Thanks, chezmoi really covers everything. Really an amazing tool
